src/units/unit.hpp

```diff
@@ -1054,6 +1054,16 @@ class unit : public std::enable_shared_from_this<unit>
 private:
 	bool resistance_filter_matches(const config& cfg, bool attacker, const std::string& damage_name, int res) const;
 
+	/**
+	 * For the provided list of resistance abilities, determine the damage resistance based on which are active and any max_value that's present.
+	 *
+	 * @param resistance_abilities A list of resistance abilities that the unit has.
```

Comment:
A list of resistance abilities affecting the unit.

Units that teach resistance to an enemy exist in Ageless, so I've checked it with a test, and this passes:
```
#wmllint: no translatables

#####
# API(s) being tested: [damage_type]alternative_type=
##
# Actions:
# Bob is a Skeleton, with resistance to blade but weakness to arcane
# Give Alice's attacks alternative_type=arcane
# Make Alice teach anti-magic, so that Bob's resistance to arcane is more than his resistance to blade
# Have Alice attack Bob
##
# Expected end state:
# Alice attacked using the blade stats, not the arcane ones.
#####
{COMMON_KEEP_A_B_UNIT_TEST "taught_resistance_with_two_attack_types" (
    [event]
        name = start

        [modify_unit]
            [filter]
                id=alice
            [/filter]
            [effect]
                apply_to=attack
                [set_specials]
                    mode=replace
                    [damage_type]
                        alternative_type=arcane
                    [/damage_type]
                [/set_specials]
            [/effect]
            [effect]
                apply_to=new_ability
                # An ability which reduces damage to both friend and foe, based on the anti-magi aura of EoMa's Matriarch of Emptiness
                # This doesn't use the TEST_ABILITY macro, because it tests add= rather than value=
                [abilities]
                    [resistance]
                        add=70
                        max_value=70
                        apply_to=fire,cold,arcane
                        affect_self=no
                        affect_allies=yes
                        affect_enemies=yes
                        [affect_adjacent]
                        [/affect_adjacent]
                        [filter_base_value]
                            less_than=70
                        [/filter_base_value]
                    [/resistance]
                [/abilities]
            [/effect]
        [/modify_unit]

        # Skeletons have base +40% vs blade, -20% vs arcane. With the +70% buff, is weaker to blade than arcane.
        {ATTACK_AND_VALIDATE 100 DAMAGE2=60}
        {SUCCEED}
    [/event]
) SIDE2_LEADER=Skeleton}
```